walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -62,24 +63,89 @@
     return meetingTimes;
   }
 
+  /** 
+   * Return an optimized version of possible meeting times that accommodates the maximum number of optional attendees. 
+   */
+  private List<TimeRange> getOptimiazedMeetingTimes(List<TimeRange> meetingTimes, List<TimeRange> unavailableTimesOptionalAttendees, long duration) {
+    List<TimeRange> optimizedMeetingTimes = new ArrayList<>();
+    
+    for (int i = 0; i < meetingTimes.size(); i++) {
+      TimeRange meetingSlot = meetingTimes.get(i);
+      boolean canSkipRange = false;
+      for (TimeRange t : unavailableTimesOptionalAttendees) {
+        int start = t.start();
+        int end = t.end();
+
+        // meeting slot is not added to optimizedMeetingtimes if optional time is the same as meeting slot 
+        // and there are other range options still available 
+        int numSlotsUnconsidered = meetingTimes.size() - i - 1;
+        if (t.equals(meetingSlot) && optimizedMeetingTimes.size() + numSlotsUnconsidered > 0) {
+          canSkipRange = true;
+          continue;
+        }
+
+        // move on to next optional time range if current t is out of range of original meeting slot 
+        if (!t.overlaps(meetingSlot) || t.contains(meetingSlot)) {
+          continue;
+        }
+
+        // update possible time range
+        List<TimeRange> possibleTimes = getPossibleTimeRanges(meetingSlot, t, duration);
+        if (possibleTimes.size() == 0) {
+          continue;
+        } else if (possibleTimes.size() == 2) {
+          optimizedMeetingTimes.add(possibleTimes.get(0));
+          meetingSlot = possibleTimes.get(1);
+        } else {
+          meetingSlot = possibleTimes.get(0);
+        }
+      }
+      // add possible time range to optimized meeting times
+      if (!canSkipRange) {
+        optimizedMeetingTimes.add(meetingSlot);
```

Comment:
is this what we want?
```
                        |----meeting----|
|---t---|

        |------------optimized:A--------|
                        |--optimized:B--|
```

It looks to me like we'll return `optimized:A`, but `optimized:A` contains time that the mandatory attendees are unavailable.

Reply:
obsolete